jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/core/JakartaForJavaAssert.java

```diff
@@ -212,9 +222,13 @@ public static void assertDiagnostics(List<Diagnostic> actual, List<Diagnostic> e
                 return simpler;
             }).collect(Collectors.toList());
         }
-        Assert.assertEquals("Unexpected diagnostics:\n" + actual, expected, received);
+        if (isDiagnosticPresent) {
```

Comment:
It is confusing when `checkForDiagnostic` actually means not to check for it. Please reverse the sense of this check so that the else part is after the if and the if part is moved to the else part. Also you will need to change the call sites (l. 164 and 173) so that it still works.